go/vt/vtgate/planbuilder/limit.go

```diff
@@ -126,7 +123,21 @@ func (l *limit) SetLimit(limit *sqlparser.Limit) error {
 		return err
 	}
 	l.elimit.Count = pv
-	l.input.SetUpperLimit(count)
+
+	switch offset := limit.Offset.(type) {
+	case *sqlparser.SQLVal:
+		pv, err = sqlparser.NewPlanValue(offset)
+		if err != nil {
+			return err
+		}
+		l.elimit.Offset = pv
+	case nil:
```

Comment:
This case can be skipped.

Reply:
Clarification: the case is needed, but it can be a no-op.

Reply:
done